### ISSUE TITLE
Add error-detection for when an attempt is made to generate an oversized figure

### ISSUE THREAD

**Issue by saxbophone** [saxbophone/libsaxbospiral#202]
The data type `sxbp_figure_t` uses 32-bit indexes and doesn't support a figure with more than 2^32 lines in it.

The rest of original issue description read:

> Because of the way that figures are generated from input data, the maximum amount of bits that a figure can be generated from is `2^32 - 1`. This is `(2^32 - 1) / 8` bytes, which is `536870911.875`, which is `536.870MB` or `511.999MiB`.
>
> In short, the function responsible for beginning figures from buffers of data, `sxbp_begin_figure()` should have a sanity-check added to check that the buffer size is not greater than this amount, otherwise problems will ensue later due to integer overflow.
>
> Although at the current speed of the library such a large figure is unlikely to be resolved (and that a figure this large would be massive when plotted out, my best guess is at roughly 40km in both directions), the library should still include this sense-check to avoid the overflow issue which could cause far worse problems.

This isn't exactly correct, for various reasons (the maths isn't correct because in the worst-case scenario (a line which moves diagonally up and right), the size will grow by 1 in each dimension every 2 lines).

So, the maximum bounds may be a bit larger and could be calculated according to this interpretation of the worst-possible line (at the shortcoming of not allowing some lines of the same length that don't go beyond the bounds for geometric reasons).

My revised calculations are like so:

- Max number of lines given worst possible line that goes up and right is `(2^32 - 1) × 2 = 8589934590`
- The maximum number of bits the user can supply for this is the same value minus `1`, so `8589934589`
- The maximum number of bytes the user can supply is this number divided by `8`, so `1073741823.625`
- To summarise: `([{2^32 - 1} × 2] - 1) ÷ 8 = 1073741823.625`
- Conveniently, this is 1 bit less than `1 GiB` of data

**Comment by saxbophone** [comment]
Closed in #206
